include/lbann/models/model.hpp

```diff
@@ -393,6 +393,14 @@ class model {
  private:
   /** Search layer graph and add all connected layers. */
   void add_connected_layers();
+  /** Insert evaluation layers where needed.
```

Comment:
It seems term layers and metric layers are collectively called evaluation layer. I have no major problem with this, but something to note if we describing LBANN concept to the "practitioner or in documenting user API (though it seems this distinction is internal to the code?)

Reply:
Right now the terminology is a little messy. A "layer term" is an objective function term and a "layer metric" is a metric; they are not layers. There is a special layer type, the "evaluation layer", that the layer term and layer metric can interact with.

In the future, I would be a fan of wiping out our objective functions and metrics and rewriting so that "objective function term" becomes identical to "layer term" and "metric" becomes identical to "layer metric".